### ISSUE TITLE
Invalid Casing on Enum Types

### ISSUE THREAD

**Issue by Stephen-Meyerhofer** [byme8/ZeroQL#35]
I'm working with an API that uses Pascal Case for its enum types. For example,

```
enum GraphDataType {
  DetailView
  SummaryView
}
```

ZeroQL is generating `GraphDataType.Detailview` and `GraphDataType.Summaryview`.

Then, when I run a query using an enum as an argument `query.Report(dataType: GraphDataType.Detailview...` I get an error such as

`Field "report" argument "dataType" requires type GraphDataType, found DETAILVIEW; Did you mean the enum value DetailView?`

How do I get ZeroQL to generate code that respects the enum casing found in the GraphQL schema?

**Comment by byme8** [comment]
Checkout 3.6.1-preview.1 version.

**Comment by byme8** [comment]
Remember that you need to update the CLI and the Nuget package.

**Comment by Stephen-Meyerhofer** [comment]
The preview version improves non-standard enums by a great deal, thank you! This API has variable casing rules on its enums. Some are all lowercase while others are Pascal case. So far it seems to work.

I discovered two follow-up issues. One is related to enums, the other is regarding custom scalars that come back as JSON. More specifically, JSON Arrays. I'll open a new issue for that one.

Enums that used reserved words as values, such as "default", make it so the generated code does not compile. Furthermore, at least one of the enums uses "Default" as a value. That one got lowercased incorrectly in the `internal static class JsonConvertersInitializers`.

To fix I did this:
1. Change "default" enum values to "@default".
2. Change ZeroQLEnumConverter instantiation arguments to ".Default" from ".default" as needed.
